### ISSUE TITLE
📈 Bản tin xu hướng AI mã nguồn mở 2026-09-06

### ISSUE THREAD

**Issue by thanhtantran** [thanhtantran/agents-radar#390]
# Xu hướng AI Mã nguồn mở 2026-09-06

> Nguồn: GitHub Trending + GitHub Search API | Thời gian tạo: 2026-09-06 02:00 UTC

---

# 📊 Báo cáo Xu hướng AI Mã nguồn mở - 06/09/2026

## 🎯 Tóm tắt hôm nay

Ngày hôm nay chứng kiến sự bùng nổ của **Agent Skills & Harness Systems** - một làn sóng mới trong cách tối ưu hóa AI agents. Các repos hàng đầu không còn là models hay frameworks mới, mà là các hệ thống "kỹ năng" giúp agents làm việc hiệu quả hơn, tư duy như senior developers, và duy trì memory xuyên sessions.

Đồng thời, **Edge AI trên RK3588** đang trở thành xu hướng nóng với nhiều công cụ deployment cho NPU hardware. Cộng đồng đang chuyển từ việc chạy models trên cloud sang chạy local trên hardware giá rẻ.

**Con số ấn tượng**: 3 repos về agent skills đạt tổng cộng **6,000+ stars trong một ngày**.

---

### 🤖 **AI Agents**

**⭐ Trending cao nhất:**

- **mattpocock/skills** (+2,692) - Skills for Real Engineers từ thư mục .agents cá nhân. Cách tiếp cận raw và thực tế cho việc cấu hình agent skills.

- **affaan-m/ECC** (+1,314 hôm nay | 249,954 total) - Agent harness performance optimization system. Tập trung vào skills, instincts, memory, security cho Claude Code, Codex, Cursor. Đây là hệ thống "meta" để tối ưu agent performance.

- **DietrichGebert/ponytail** (+2,845 hôm nay | 128,001 total) - Khiến AI agent "nghĩ như senior dev lười nhất" - triết lý "code tốt nhất là code không viết". Phản ánh xu hướng agents cần học cách tránh over-engineering.

- **NousResearch/hermes-agent** (+575 hôm nay | 242,037 total) - "The agent that grows with you" - agent có khả năng self-evolution và learning.

**Vertical agents:**

- **Panniantong/Agent-Reach** (78,247 ⭐) - Agent có "mắt để thấy internet" - scrape Twitter, Reddit, YouTube, GitHub, Bilibili, XHS, zero API fees.

- **career-ops-hq/career-ops** (70,244 ⭐) - AI job search agent: scan job portals, đánh giá A-H + score 1-5, tailor CV, track applications.

- **ZhuLinsen/daily_stock_analysis** (64,667 ⭐) - LLM-powered stock analysis với multi-market data, real-time news, decision dashboard.

- **hugohe3/ppt-master** (52,221 ⭐) - Tự động tạo PowerPoint từ documents/topics với native shapes, transitions, animations, charts, audio narration.

- **zhayujie/CowAgent** (46,785 ⭐) - Super AI assistant & Agent Harness. Plans tasks, chạy tools/skills, self-evolves với memory và knowledge.

**Multi-agent & Orchestration:**

- **ruvnet/ruflo** (+136) - Agent meta-harness gốc. Deploy multi-player swarms, coordinate autonomous workflows với adaptive memory, self-learning, RAG integration.

### 🔧 **AI Infrastructure**

**Agent Development Tools:**

- **anthropics/skills** (+475) - Public repository chính thức cho Agent Skills từ Anthropic.

- **humanlayer/skills** (+442) - TypeScript-based agent skills repository.

- **WorldFlowAI/everything-claude-code** (+95) - Claude Code toolkit đầy đủ: agents, commands, skills, rules, hooks.

- **anomalyco/opencode** (+725) - Open source coding agent.

**Memory & Context:**

- **thedotmack/claude-mem** (93,297 ⭐) - Persistent Context Across Sessions cho mọi agent. Capture session data, compress với AI, inject lại context. Universal support.

- **mem0ai/mem0** (64,748 ⭐) - The Memory Layer for AI Agents - Drop-in memory infrastructure cho production.

- **jaylfc/taosmd** (77 ⭐) - Local-first AI memory chạy offline trên machine 8GB+ RAM với zero-loss archive, knowledge graph, hybrid retrieval.

**Compression & Optimization:**

- **headroomlabs-ai/headroom** (69,042 ⭐) - Compress tool outputs, logs, files, RAG chunks trước khi đến LLM. 20% fewer tokens cho coding agents, 60-95% cho JSON.

**Inference & Serving:**

- **magnitudedev/magnitude** (+674) - Open source inference server chạy best local models, plug vào agent bạn đang dùng (Pi, OpenCode, Hermes, Codex, Claude Code, Cline).

- **ollama/ollama** (180,253 ⭐) - Chạy Kimi-K2.6, GLM-5.2, MiniMax, DeepSeek, gpt-oss, Qwen, Gemma locally.

**Web & Context:**

- **firecrawl/firecrawl** (176,920 ⭐) - Context API để search, scrape, interact với web at scale.

### 🧠 **Models & Training**

- **huggingface/transformers** (164,837 ⭐) - State-of-the-art ML models framework cho text, vision, audio, multimodal.

### 📦 **AI Applications**

**Platforms & Studios:**

- **open-webui/open-webui** (151,066 ⭐) - User-friendly AI Interface hỗ trợ Ollama, OpenAI API.

- **CherryHQ/cherry-studio** (51,480 ⭐) - AI productivity studio với smart chat, autonomous agents, 300+ assistants.

- **langgenius/dify** (154,541 ⭐) - Build Agentic workflows, RAG pipelines với rich AI model/tool support. Deploy cloud, VPC, hoặc self-hosted.

- **Mintplex-Labs/anything-llm** (65,662 ⭐) - "Stop renting your intelligence. Own it." - Powerful local-first agent experience.

**Autonomous Systems:**

- **Significant-Gravitas/AutoGPT** (187,162 ⭐) - Vision of accessible AI for everyone.

- **Hmbown/Codewhale** (40,914 ⭐) - Open-source coding agent cho terminal, built in Rust, community-driven.

**Specialized Apps:**

- **harry0703/MoneyPrinterTurbo** (120,872 ⭐) - AI workflow tạo HD short videos từ topic/keyword.

- **siyuan-note/siyuan** (46,184 ⭐) - Privacy-first, self-hosted knowledge workspace cho humans và AI agents.

**Frontend Frameworks:**

- **CopilotKit/CopilotKit** (37,213 ⭐) - The Frontend Stack for Agents & Generative UI. React, Angular, Mobile, Slack. Makers of AG-UI Protocol.

### 🔍 **RAG & Knowledge**

**RAG Engines:**

- **infiniflow/ragflow** (90,102 ⭐) - Leading RAG engine kết hợp cutting-edge RAG với Agent capabilities.

- **Graphify-Labs/graphify** (115,086 ⭐) - Turn any codebase (docs, SQL, configs, PDFs) thành queryable knowledge graph. Local deterministic AST parsing, mọi edge được giải thích, no vector store.

**Frameworks:**

- **langchain-ai/langchain** (145,722 ⭐) - The agent engineering platform.

- **Shubhamsaboo/awesome-llm-apps** (136,246 ⭐) - 100+ AI Agents, Agent Skills và RAG Apps - Free và Open Source.

**Education:**

- **datawhalechina/hello-agents** (77,146 ⭐) - 《从零开始构建智能体》 - Tutorial xây dựng agent từ đầu.

**Lightweight Solutions:**

- **HKUDS/nanobot** (47,736 ⭐) - Ultra-lightweight, self-hosted personal AI agent framework với WebUI, tools, memory, MCP, multi-agent workflows.

### 🔌 **Embedded AI**

**🔥 Hot trend: RK3588 NPU Deployment**

**RKLLM (Large Language Models on Rockchip):**

- **Leon6225/InternVL3.5-4B-NPU** (5 ⭐) - InternVL3.5-4B cho RK3588 NPU với multimodal vision-language understanding.

- **Ben1332/qwen3.8-27b-rkllm-rk3588-nanopi-m6** (1 ⭐) - Technical proof: convert và run Qwen3.8-27B as RKLLM W8A8 trên NanoPi M6 (RK3588S).

- **tristanpenman/vlm-rknn** (1 ⭐) - Starter CMake project cho vision-language models trên Rockchip devices via RKNN/RKLLM.

- **AKHYui/rkllama-webui** - Web UI cho RKLLM NPU models với multi-session chat, SSE streaming, model mounting, RAG (bge-small-zh + ChromaDB).

- **ambagesthickskin162/Qwen3.5-4B-NPU** - Deploy Qwen3.5-4B trên NPU hardware với efficient local inference.

**RKNPU (Rockchip Neural Processing Unit):**

- **jaylfc/taOS** (514 ⭐) - Self-hosted AI agent OS. Memory, chat, agents, files stay on hardware you own, offline by default. Offline AI memory, web desktop + app store, auto-clustering across consumer hardware (Orange/Raspberry Pi, Mac mini, gaming PC).

- **isac322/rkmon** (9 ⭐) - Real-time hardware monitor TUI cho RK3588 SBCs. Như htop nhưng cho GPU, NPU, VPU, RGA, thermal zones.

- **oRKLLM/ork-driver** (3 ⭐) - Clean-room userspace matmul library cho Rockchip NPU.

- **YeWenxuan64/Edge_Inferencer** (2 ⭐) - Unified edge AI inference engine - một Python API cho Rockchip NPU, Qualcomm HTP & ONNX Runtime.

- **YeWenxuan64/Edge_ModelDeploy** - CV model deployment toolchain: PyTorch/TensorFlow → ONNX → quantize & deploy to Rockchip RKNPU & Qualcomm HTP.

**Hardware Support:**

- **freed-dev-llc/terraform-provider-turingpi** (7 ⭐) - Terraform provider cho Turing Pi 2.5 BMC cluster deployment.

- **gregordinary/patches** (4 ⭐) - Out-of-tree patches cho Rockchip RK3588 mainline NPU driver và HW video-transcode.

**Orange Pi Ecosystem:**

- **MichaIng/DietPi** (6,244 ⭐) - Lightweight OS cho single-board computers.

- **RaspAP/raspap-webgui** (5,218 ⭐) - Full-featured wireless router setup cho Debian-based devices.

- **Ponce1969/contador-oriental-ai** (3 ⭐) - Sistema gestión financiera familiar với AI local. Python, Fleting, PostgreSQL, Ollama. 100% offline, ready cho Orange Pi 5 Plus.

- **CERALIVE/image-building-pipeline** (3 ⭐) - mkosi-based image builder cho CeraLive streaming appliances - produces RAUC OTA packages cho RK3588 devices.

**Autonomy & Robotics:**

- **LingZhen07/ros2-stm32-autonomous-robot** (3 ⭐) - ROS 2 + STM32 autonomous robot với CAN FD drivetrain, RPLIDAR, SLAM, navigation trên Orange Pi AI Pro.

- **dustbowlair424/NavCore-Pixhawk** (7 ⭐) - Real-time GPS-denied INS navigation cho drones dùng Pixhawk Cube Orange và Raspberry Pi 4.

### 🎨 **Content & Design**

- **cathrynlavery/diagram-design** (+855) - 38 editorial diagram types cho Claude Code, Codex, Pi. Self-contained HTML + SVG, no shadows, no Mermaid slop.

- **blader/humanizer** (+990) - Agent skill loại bỏ dấu hiệu AI-generated writing.

### 🛡️ **Security & Tools**

- **bikini/exploitarium** (+230) - Archive of public exploit PoCs và vulnerability research writeups. Educational security resource.

- **BraveOPotato/FckSignups** (+68) - List of open-source, in-browser tools không cần signup.

- **fmtlib/fmt** (+134) - Modern C++ formatting library.

- **nvm-sh/nvm** (+22) - Node Version Manager.

---

### 1. **Agent Skills Revolution** 🚀
Không còn là "model nào mạnh hơn", mà giờ là "agent nào được trang bị skills tốt hơn". Top 3 repos về skills thu hút **6,000+ stars trong một ngày** cho thấy cộng đồng đang chuyển focus từ model capabilities sang agent optimization.

**Pattern mới:**
- Skills như những "công cụ" có thể tái sử dụng
- Agent harnesses để orchestrate và optimize
- Triết lý "làm ít hơn, làm đúng hơn" (ponytail: "code tốt nhất là code không viết")

### 2. **Persistent Memory = Must-Have** 🧠
`claude-mem` (93K stars) và các memory systems khác cho thấy agents không thể chỉ là stateless. Memory xuyên sessions là điều kiện để agents trở nên truly useful.

**Xu hướng:**
- Local-first memory (taosmd)
- Compression-first context management (headroom: 60-95% token reduction)
- Knowledge graphs thay vì pure vector stores (graphify)

### 3. **Edge AI on RK3588 Exploding** 💥
Đột nhiên có **hơn 10 repos mới** về RKLLM và RKNPU deployment trong tuần này. Cộng đồng đang nghiêm túc với việc chạy LLMs trên $100 SBCs.

**Drivers:**
- Hardware giá rẻ với NPU mạnh (RK3588)
- Models nhỏ hơn nhưng vẫn capable (Qwen3.5-4B, InternVL3.5-4B)
- Tooling đang mature (RKLLM, quantization tools)
- Privacy concerns → local inference

### 4. **Meta-Systems & Orchestration** 🎭
ECC, ruflo, CowAgent - các "systems of systems" để manage và optimize agents. Không chỉ là một agent, mà là framework để agents grow, learn, collaborate.

### 5. **Vertical AI Applications Maturing** 📊
Từ job search (career-ops), stock analysis (daily_stock_analysis), PPT generation (ppt-master) đến financial management (contador-oriental-ai) - AI agents đang được applied vào specific domains với serious traction.

### 6. **Privacy-First, Self-Hosted Wave** 🔒
Nhiều repos nhấn mạnh "self-hosted", "offline by default", "privacy-first":
- taOS: "Your data stays on hardware you own"
- anything-llm: "Stop renting your intelligence"
- siyuan: "privacy-first knowledge workspace"

Đây là phản ứng với concerns về data privacy và vendor lock-in.

### 7. **Compression & Efficiency Focus** ⚡
headroom (69K stars) với claim "20-60% token reduction" cho thấy optimization không chỉ là về model size, mà còn về làm sao giảm context bloat. Trong thời đại context windows lớn, paradoxically việc compress data lại trở nên quan trọng.

---

### 🏆 **Most Discussed: Agent Skills Ecosystem**

Community đang debate:
- **Standardization vs Flexibility**: Có nên có một "standard skill format" không?
- **Security concerns**: Skills có thể execute code - làm sao ensure safety?
- **Discovery problem**: Với hàng trăm skills, làm sao agents biết dùng cái nào?

mattpocock's approach (share .agents directory directly) được yêu thích vì **pragmatic và transparent**.

### 🔥 **Hottest Debate: Local vs Cloud**

Hai진영 rõ ràng:
- **Team Local**: RK3588 community, taOS, anything-llm - "own your intelligence"
- **Team Cloud**: Dify, larger model users - "scale và convenience matter"

Thực tế: **Hybrid winning** - local cho sensitive/offline, cloud cho heavy lifting.

### 🎯 **Attention Magnet: ponytail Philosophy**

"Makes your agent think like the laziest senior dev" - controversial nhưng resonates mạnh. +2,845 stars trong một ngày vì nó capture một truth:
- Over-engineering là problem #1
- Agents nên learn "when NOT to code"
- Simplicity > complexity

### 🌊 **Rising Interest: Edge AI Democratization**

RK3588 community đang build momentum. Từ niche hardware sang serious ecosystem với:
- Inference servers (magnitude)
- Monitoring tools (rkmon)
- Driver development (ork-driver)
- Web UIs (rkllama-webui)
- Complete OS solutions (taOS)

**Prediction**: 6-12 tháng nữa, chạy LLMs trên $100 hardware sẽ trở thành mainstream.

---

## 📌 Key Takeaways

1. **Skills > Models**: Cộng đồng đang optimize how agents use tools, không chỉ tools là gì.

2. **Memory is Infrastructure**: Persistent, compressed, structured memory đang trở thành core requirement.

3. **Edge AI is Real**: RK3588 + RKLLM không còn là experiment, đang là production-ready option.

4. **Privacy Matters**: Self-hosted, local-first solutions đang gain serious traction.

5. **Vertical AI Apps**: Specific-domain applications (jobs, stocks, presentations) với real users.

6. **Meta-Orchestration**: Frameworks để manage frameworks - agents managing agents.

Ngày 06/09/2026 đánh dấu **shift từ "powerful models" sang "smart agent systems"**. 🚀

---
*Bản tin này được tạo tự động bởi [agents-radar](https://github.com/thanhtantran/agents-radar).*

**Comment by x711criptic** [comment]
x711 works in Cline as an MCP server — 41 tools (web search, price feeds, shared Hive memory, tx sim), one config line:

```json
{"mcpServers":{"x711":{"type":"streamable-http","url":"https://x711.io/mcp"}}}
```

Free tier, 10 calls/day, no signup.

_Feel free to ignore if this isn't relevant to what you're working on._

---
**[⚡ x711.io/go](https://x711.io/go?source=gh-issue)** — 7 seconds, no terminal. Free tier, 41 tools, MCP included.
